### DISCUSSION TITLE
issue: `append` adds duplicate fields to field array when run inside `useEffect`

### DISCUSSION THREAD

**Discussion by Sreejit7** [react-hook-form/react-hook-form#9592] [Q&A] [answered]
I'm using `useFieldArray` to generate an array of fields.
Inside `useEffect`, I'm looping over an array to `append` fields to the array like so:

```jsx
useEffect(() => {
    const parsedBiller: BillerResponse = JSON.parse(biller)
  
    // Append fields to form
    parsedBiller.customerParams.forEach(customerParam => {
      append({
        name: customerParam.paramName,
        value: ""
      });
    });
}, []);
  ```
  
I'm using a setup like this:

```jsx
  interface BillerForm {
    customerParams: {
      name: string;
      value: string;
    }[];
  }

  const { register, control, handleSubmit } = useForm<BillerForm>();
  const { append, fields } = useFieldArray({ name: "customerParams", control, shouldUnregister: true });
```

And rendering the below:

```jsx
     {customerParamFields.map((field, index) =>
        <Row key={field.id}>
            <InputField
                label={`Enter ${field.name}`}
                {...register(`customerParams.${index}.name` as const)}
             />
          </Portion>
        </Row>
      )}
 ```

What I see on the UI:

![image](https://user-images.githubusercontent.com/52678249/207811871-3234aff5-2556-4c21-a23b-3b274f7fe79e.png)

What am I doing wrong? Or am I missing something? Please help. 🙏🏼

**Comment by Moshyfawn** [comment, marked as answer]
With `React.StrictMode`, the effect is ran twice; it's not the place to `append` your fields in. Consider using `defaultValues` or `useEffect` + `reset` if you have async data coming in after the form's been init'ed

**Reply by Sreejit7** [reply]
Thanks, using `useEffect` + `reset` did resolve the issue.
